dev/forbidden_from_release_tests/bin/main.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart=2.9
+// TODO(dnfield): migrate this once vm_snapshot_analysis is migrated.
+// https://github.com/dart-lang/sdk/issues/45683
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+import 'package:vm_snapshot_analysis/program_info.dart';
+import 'package:vm_snapshot_analysis/v8_profile.dart';
+
+import 'package:meta/meta.dart';
+
+const ProcessManager processManager = LocalProcessManager();
+const FileSystem fs = LocalFileSystem();
+
+Future<void> main(List<String> args) async {
+  final Options options = Options.fromArgs(args);
+
+  final Directory tempDirectory = fs.systemTempDirectory.createTempSync('forbidden_imports');
+  print('Using $tempDirectory for temporary files.');
+
+  final List<String> command = <String>[
+    options.flutter.path,
+    'build',
+    'apk',
+    '--target-platform',
+    'android-arm64',
+    '--release',
+    '--analyze-size',
+    '--code-size-directory',
+    tempDirectory.path,
+    '-v',
+  ];
+
+  await _runStreamed(command, workingDirectory: options.target.parent.path);
+
+  final File v8SnapshotInfo = tempDirectory.childFile('snapshot.arm64-v8a.json');
+
+  final String json = v8SnapshotInfo.readAsStringSync();
+  final Snapshot snapshot = Snapshot.fromJson(jsonDecode(json) as Map<String, dynamic>);
+  final ProgramInfo programInfo = toProgramInfo(snapshot);
+
+  final List<String> foundForbiddenTypes = <String>[];
+  bool fail = false;
+  for (final String forbiddenType in options.forbiddenTypes) {
+    final int slash = forbiddenType.indexOf('/');
+    final int doubleColons = forbiddenType.indexOf('::');
+    if (slash == -1 || doubleColons < 2) {
+      print('Invalid forbidden type "$forbiddenType". The format must be <package_uri>::<type_name>, e.g. package:flutter/src/widgets/framework.dart::Widget');
+      fail = true;
+      continue;
+    }
+
+    final List<String> lookupPath = <String>[
+      forbiddenType.substring(0, slash),
+      forbiddenType.substring(0, doubleColons),
```

Comment:
Is this expected that they both start at 0? Should the second one start after the slash?

Reply:
Yes, expected - we want `package:flutter/src/whatever/whatever.dart`, not `src/whatever/whatever.dart` when we do the typelookup.